docs/user-guide/slicing.ipynb

```diff
@@ -687,6 +687,134 @@
+    "\n",
+    "The condition variable must be 1-D and must be compatible with the shape of the indexed object.\n",
+    "Consider a 2-D variable:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import scipp as sc\n",
+    "\n",
+    "var = sc.arange('dummy', 12).fold(dim='dummy', sizes={'x':6, 'y':2})\n",
+    "var"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Indexing with a boolean variable corresponds to extracting rows or columns:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "condition = sc.array(dims=['x'], values=[True, False, False, True, False, False])\n",
+    "var[condition]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "condition = sc.array(dims=['y'], values=[False, True])\n",
+    "var[condition]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "**Note**\n",
+    "\n",
+    "By necessity &mdash; since the returned elements are generally non-contiguous &mdash; indexing with a condition variable returns a *copy* of the input data.\n",
+    "This is in contrast to [positional indexing](#Positional-indexing) and [label-based indexing](#Label-based-indexing) which return a view.\n",
+    "\n",
+    "Note that indexing with a condition variable *always* returns a copy, even if the selected elements for a contiguous range.\n",
+    "    \n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Given a multi-dimensional condition variable such as"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "condition = var < 5\n",
+    "condition"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "an indexing attempt will raise an error:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "raises-exception"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "var[condition]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Unlike [numpy's boolean array indexing](https://numpy.org/doc/stable/user/basics.indexing.html#boolean-array-indexing) scipp does not support this, since it would require automatic flattening for the output, which is incompatible with scipp's philosophy of enforcing labeled dimensions.\n",
+    "If such indexing is required, flatten by hand instead:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "var.flatten(to='elem')[condition.flatten(to='elem')]"
+   ]
```

Comment:
Can you add something about bin edges? Is is possible to use masks with bin-edges?

Reply:
Done.